### PR TITLE
[2077] Add borderStyle to container

### LINKED ISSUE #2077: Add borderStyle for containers group and flexbox

### PR DESCRIPTION
**Pull request by frouene**
Bug: https://github.com/eclipse-sirius/sirius-components/issues/2077

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [x] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

**Review by sbegaudeau — DISMISSED** [review]
- No need for a "." at the end of your commit message
- The update of the CHANGELOG is missing

(I have not reviewed anything in there apart from that, feel free to remove my review once those issues are fixed)

File: packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/ContainerBorderStyle.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+        public Builder lineStyle(ContainerBorderLineStyle lineStyle) {
+            this.lineStyle = Objects.requireNonNull(lineStyle);
+            return this;
+        }
+
+        public ContainerBorderStyle build() {
+            ContainerBorderStyle borderStyle = new ContainerBorderStyle();
+            borderStyle.color = this.color; // Optional on purpose
+            borderStyle.radius = Objects.requireNonNull(this.radius);
```
Comment (AxelRICHARD) [inline packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/ContainerBorderStyle.java]: `radius` is an int so no need to put a `Objects.requireNonNull(...)` here.

File: packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/ContainerBorderStyle.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            borderStyle.size = Objects.requireNonNull(this.size);
```
Comment (AxelRICHARD) [inline packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/ContainerBorderStyle.java]: `size` is an int so no need to put a `Objects.requireNonNull(...)` here.

File: packages/forms/frontend/sirius-components-forms/src/form/FormEventFragments.types.ts [hunk 3]
```diff
@@ -111,9 +111,17 @@ export interface GQLGroup {
   displayMode: GQLGroupDisplayMode;
   toolbarActions: GQLToolbarAction[];
   widgets: GQLWidget[];
+  borderStyle: GQLContainerBorderStyle;
   __typename: string;
 }
 
+export interface GQLContainerBorderStyle {
+  color: string;
```
Comment (AxelRICHARD) [inline packages/forms/frontend/sirius-components-forms/src/form/FormEventFragments.types.ts]: Shouldn't it be `string | null`?

File: packages/forms/frontend/sirius-components-forms/src/groups/Group.tsx [hunk 4]
```diff
@@ -11,20 +11,25 @@
...
-const useGroupStyles = makeStyles((theme) => ({
+const useGroupStyles = makeStyles<Theme, GroupStyleProps>((theme) => ({
   group: {
     display: 'flex',
     flexDirection: 'column',
-    paddingTop: theme.spacing(1),
+    margin: theme.spacing(1),
+    padding: theme.spacing(0.5),
+    borderWidth: ({ borderSize }) => borderSize || 0,
+    borderColor: ({ borderColor }) => borderColor || 'black',
```
Comment (AxelRICHARD) [inline packages/forms/frontend/sirius-components-forms/src/groups/Group.tsx]: So you change the default behavior to no visible border to black border, even if the specifier set `None` to the border color? Unless I'm missing something here, when no border color is specified, `transparent` should be used.

File: packages/forms/frontend/sirius-components-forms/src/propertysections/FlexboxContainerPropertySection.tsx [hunk 5]
```diff
@@ -20,6 +20,14 @@ import { PropertySectionLabel } from './PropertySectionLabel';
 
 const useFlexboxContainerPropertySectionStyles = makeStyles<Theme, FlexboxContainerPropertySectionStyleProps>(
   (theme) => ({
+    containerAndLabel: {
+      margin: theme.spacing(0.5),
+      padding: theme.spacing(0.5),
+      borderWidth: ({ borderSize }) => borderSize || 0,
+      borderColor: ({ borderColor }) => borderColor || 'black',
```
Comment (AxelRICHARD) [inline packages/forms/frontend/sirius-components-forms/src/propertysections/FlexboxContainerPropertySection.tsx]: Same comment

**Review by AxelRICHARD — CHANGES_REQUESTED** [review]

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**